inst/include/state_saver.hpp

```diff
@@ -83,35 +118,42 @@ class StateSaver {
   }
 
 
-  void save_state(const State<S, real_type> &state, int current_year) {
+  void save_state(const State<ModelVariant, real_type> &state, int current_year) {
```

Comment:
Split this state saver on the ModelVariant too. And have this call the save_state from the base and child StateSaver objects